reader/tests/system/test_client.py

```diff
@@ -33,3 +42,153 @@ def test_invalid_requests(json_client):
     for request in requests:
         response = json_client.post(Route.GET.URL, request)
         assert_error_response(response, ERROR_CODES.INVALID_REQUEST)
+
+
+class TestConcurrentRequests:
+    """
+    The database methods are patched and are just locking two locks and releasing them again.
+    The first is to indicate that the handling of the request was started, the second to lock the
+    execution from "outside". Procedure of the events:
+
+    +-------------+---------------------+------------------------+------------------------+---------------------+
+    |    main     |         T1          |           T2           |           T3           |         T4          |
+    +-------------+---------------------+------------------------+------------------------+---------------------+
+    | L1 locked   |                     |                        |                        |                     |
+    | T1 started  |                     |                        |                        |                     |
+    |             | request sent        |                        |                        |                     |
+    |             | connection acquired |                        |                        |                     |
+    |             | I1 locked           |                        |                        |                     |
```

Comment:
Whats the purpose of the `I` locks, if they are used in just one thread?

Reply:
> [...] to indicate that the handling of the request was started

The indicator locks are checked in the test to assert that the request is waiting because of the second lock and not because no connection is available